shells/android/java/arcs/crdt/parcelables/ParcelableCrdtEntity.kt

```diff
@@ -0,0 +1,292 @@
+            override fun createFromParcel(parcel: Parcel): ReferenceImpl {
+                val id = requireNotNull(parcel.readString(), {
+                    "ID not found in parcel when reading ParcelableCrdtEntity.ReferenceImpl"
+                })
+                return ReferenceImpl(CrdtEntity.ReferenceImpl(id))
+            }
+
+            override fun newArray(size: Int): Array<ReferenceImpl?> = arrayOfNulls(size)
+        }
+    }
+
+
+    /** Parcelable variant of [CrdtEntity.Data]. */
+    data class Data(
+        override val actual: CrdtEntity.Data
+    ) : ParcelableCrdtData<CrdtEntity.Data> {
+        override var versionMap = actual.versionMap
+
+        @Suppress("UNCHECKED_CAST")
+        override fun writeToParcel(parcel: Parcel, flags: Int) {
+            parcel.writeTypedObject(actual.versionMap.toParcelable(), flags)
+
+            parcel.writeInt(actual.singletons.size)
+            actual.singletons.forEach { (field, value) ->
+                parcel.writeString(field)
+                parcel.writeTypedObject(ParcelableCrdtSingleton.Data(
+                    value.data as CrdtSingleton.Data<Referencable>), flags
+                )
+            }
+
+            parcel.writeInt(actual.collections.size)
+            actual.collections.forEach { (field, value) ->
+                parcel.writeString(field)
+                parcel.writeTypedObject(
+                    ParcelableCrdtSet.Data(value.data as CrdtSet.Data<Referencable>), flags)
+            }
+        }
+
+        override fun describeContents(): Int = 0
+
+        companion object CREATOR : Parcelable.Creator<Data> {
+            @Suppress("UNCHECKED_CAST")
+            override fun createFromParcel(parcel: Parcel): Data {
+                val versionMap = requireNotNull(
+                    parcel.readTypedObject(ParcelableVersionMap.CREATOR)
+                ) { "No VersionMap found in parcel when reading ParcelableCrdtEntity.Data" }
+
+                val singletons = mutableMapOf<FieldName, CrdtSingleton<CrdtEntity.Reference>>()
+                val numSingletons = parcel.readInt()
+                repeat(numSingletons) {
+                    val field = requireNotNull(parcel.readString())
+                    val data = requireNotNull(
+                        parcel.readTypedObject(ParcelableCrdtSingleton.Data.CREATOR)
+                    ).actual
+                    singletons[field] = CrdtSingleton.createWithData(
+                        data as CrdtSingleton.Data<CrdtEntity.Reference>
+                    )
+                }
+
+                val collections = mutableMapOf<FieldName, CrdtSet<CrdtEntity.Reference>>()
+                val numCollections = parcel.readInt()
+                repeat(numCollections) {
+                    val field = requireNotNull(parcel.readString())
+                    val data = requireNotNull(
+                        parcel.readTypedObject(ParcelableCrdtSingleton.Data.CREATOR)
+                    ).actual
```

Comment:
Same suggestion re: `.actual` within the `requireNotNull`

Reply:
Done